src/hook/usePushNotificationsPermissions.ts

```diff
@@ -0,0 +1,46 @@
+import {useQueryClient, useQuery, useMutation} from 'react-query';
+import messaging, {FirebaseMessagingTypes} from '@react-native-firebase/messaging';
+import {usePushTopics} from './usePushTopics';
+
+export function usePushNotificationsPermissions() {
+  const {subscribeToAllTopics} = usePushTopics();
+  const queryClient = useQueryClient();
+
+  const {pushAuthorizationStatus} = usePushAuthorizationStatus();
```

Comment:
@franjs I changed the previous logic to use the  `usePushAuthorizationStatus`. it's not using  useAsync there but that's because I like the invalidate it after the mutation so user would be automatically moved to next route.